base/support_menus/MainStatusDjinn.ts

```diff
@@ -0,0 +1,226 @@
+        return;
+    }
+
+    public on_left() {
+        if (this.djinn_pages.length <= 1) return;
+
+        this.current_page = (this.current_page + this.djinn_pages.length - 1) % this.djinn_pages.length;
+        this.load_page();
+    }
+
+    public on_right() {
+        if (this.djinn_pages.length <= 1) return;
+
+        this.current_page = (this.current_page + 1) % this.djinn_pages.length;
+        this.load_page();
+    }
+
+    public on_up() {
+        return;
+    }
+
+    public on_down() {
+        return;
+    }
+
+    public initialize() {
+        this.djinn_group.x = this.game.camera.x;
+        this.djinn_group.y = this.game.camera.y;
+
+        this.make_pages();
+        this.window.page_indicator.initialize(this.djinn_pages.length, this.current_page);
+
+        this.current_page = 0;
+        this.load_page();
+    }
+
+    private load_page() {
+        this.clear();
+        this.window.page_indicator.select_page(this.current_col);
+
+        const djinn = this.djinn_pages[this.current_page];
+
+        for (let i = 0; i < ordered_elements.length; i++) {
+            const elem = ordered_elements[i];
+            const elem_djinn: Djinn[] = djinn[ordered_elements[i]];
+
+            for (let n = 0; n < elem_djinn.length; n++) {
+                let x_pos = MainStatusDjinn.STARS.X + i * MainStatusDjinn.STARS.X_SHIFT;
+                let y_pos = MainStatusDjinn.STARS.Y + n * MainStatusDjinn.STARS.Y_SHIFT;
+
+                const star = this.window.create_at_group(
+                    x_pos,
+                    y_pos,
+                    ordered_elements[i] + "_star",
+                    undefined,
+                    undefined,
+                    MainStatusDjinn.GROUP_KEY
+                );
+                this.state_sprites.push(star);
+
+                x_pos = MainStatusDjinn.NAMES.X + i * MainStatusDjinn.NAMES.X_SHIFT;
+                y_pos = MainStatusDjinn.NAMES.Y + n * MainStatusDjinn.NAMES.Y_SHIFT;
+
+                const name = this.window.set_text_in_position(
+                    elem_djinn[n].name,
+                    x_pos,
+                    y_pos,
+                    false,
+                    false,
+                    undefined,
+                    false,
+                    MainStatusDjinn.GROUP_KEY
+                );
+                this.state_sprites.push(name.text, name.shadow);
+            }
+
+            const x_pos = MainStatusDjinn.SPRITES.CENTER_X + i * MainStatusDjinn.SPRITES.SHIFT;
+            let y_pos = MainStatusDjinn.SPRITES.Y;
+
+            const djinni_sprite = this.djinn_group.create(x_pos, y_pos, elem + "_djinn_set");
+
+            djinni_sprite.anchor.setTo(0.5, 1.0);
+            djinni_sprite.scale.x = -1;
+
+            const direction = "down";
+            const action = "set";
+
+            this.data.info.djinni_sprites[elem].setAnimation(djinni_sprite, action);
+            djinni_sprite.animations.play(action + "_" + direction);
```

Comment:
Please use `this.data.info.djinni_sprites[elem].getAnimationKey` instead of `action + "_" + direction`.